packages/xo-server/src/xapi-object-to-xo.js

```diff
@@ -480,6 +499,14 @@ const TRANSFORMS = {
           : link(obj.$PBDs[0], 'host'),
       $PBDs: link(obj, 'PBDs'),
     }
+
+    Object.entries(TYPES_BY_ALLOCATION).forEach(([key, types]) => {
```

Comment:
The loop continue even if we found the type. I think using [`findKey`](https://devdocs.io/lodash~4/index#findKey) will be more appropriate.